latex/chain.tex

```diff
@@ -0,0 +1,787 @@
+\section{Blockchain layer}
+\label{sec:chain}
+
+\newcommand{\leteq}{\ensuremath{\mathrel{\mathop:}=}}
+
+\newcommand{\Seed}{\type{Seed}}
+\newcommand{\seedOp}{\star}
+\newcommand{\Proof}{\type{Proof}}
+\newcommand{\Seedl}{\mathsf{Seed_\ell}}
+\newcommand{\Seedn}{\mathsf{Seed_n}}
+\newcommand{\slotToSeed}[1]{\fun{slotToSeed}~ \var{#1}}
+
+\newcommand{\Bool}{\type{Bool}}
+\newcommand{\T}{\type{T}}
+\newcommand{\vrf}[3]{\fun{vrf}_{#1} ~ #2 ~ #3}
+\newcommand{\verifyVrf}[4]{\fun{verifyVrf}_{#1} ~ #2 ~ #3 ~#4}
+
+\newcommand{\HashBlock}{\type{HashBlock}}
+\newcommand{\hashBlock}[1]{\fun{hashBlock}~ \var{#1}}
+\newcommand{\BHeader}{\type{BHeader}}
+\newcommand{\BHBody}{\type{BHBody}}
+
+\newcommand{\header}[1]{\fun{header}~\var{#1}}
+\newcommand{\hbody}[1]{\fun{hbody}~\var{#1}}
+\newcommand{\hsig}[1]{\fun{hsig}~\var{#1}}
+\newcommand{\bprev}[1]{\fun{bprev}~\var{#1}}
+\newcommand{\bhash}[1]{\fun{bhash}~\var{#1}}
+\newcommand{\bsig}[1]{\fun{bsig}~\var{#1}}
+\newcommand{\bissuer}[1]{\fun{bissuer}~\var{#1}}
+\newcommand{\bseedl}[1]{\fun{bseed}_{\ell}~\var{#1}}
+\newcommand{\bprfn}[1]{\fun{bprf}_{n}~\var{#1}}
+\newcommand{\bseedn}[1]{\fun{bseed}_{n}~\var{#1}}
+\newcommand{\bprfl}[1]{\fun{bprf}_{\ell}~\var{#1}}
+
+\newcommand{\BHeaderEnv}{\type{BHeaderEnv}}
+\newcommand{\NewEpochEnv}{\type{NewEpochEnv}}
+\newcommand{\NewEpochState}{\type{NewEpochState}}
+\newcommand{\BBodyEnv}{\type{BBodyEnv}}
+\newcommand{\BBodyState}{\type{BBodyState}}
+\newcommand{\ChainEnv}{\type{ChainEnv}}
+\newcommand{\ChainState}{\type{ChainState}}
+\newcommand{\ChainSig}{\type{ChainSig}}
+
+
+\begin{note}
+  This section is a work in progress.
+  See \cite{ouroboros_praos}.
+\end{note}
+
+\subsection{Verifiable Random Functions (VRF)}
+\label{sec:defs-vrf}
+
+\begin{figure}[htb]
+  %
+  \emph{Abstract types}
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \var{seed} & \Seed  & \text{seed for pseudo-random number generator}\\
+      \var{prf} & \Proof  & \text{VRF proof}\\
+    \end{array}
+  \end{equation*}
+  %
+  \emph{Seed operation}
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \seedOp & \Seed \to \Seed \to \Seed & \text{binary seed operation}
+    \end{array}
+  \end{equation*}
+  %
+  \emph{Abstract VRF functions}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \vrf{\T}{}{} & \SKey \to \Seed \to \T\times\Proof
```

Comment:
what is `T`?

Reply:
a generic type. I should probably mention that somehow...